spec/ssh_scan/result_spec.rb

```diff
@@ -212,4 +215,22 @@
     end
   end
 
+  context "when dealing with errors" do 
+    it "should append errors " do
+      compliance = {
+        :policy => "Test Policy",
+        :compliant => true,
+        :recommendations => ["do this", "do that"],
+        :references => ["https://reference.example.com"],
+      }
+      result = SSHScan::Result.new()
+      result.set_compliance = compliance
+      result.error = "This is an error"
+
+      expect(result.to_hash).to be_kind_of(Hash)
+      expect(result.to_hash["error"]).to eql("This is an error")
```

Comment:
I believe this was the case that would have triggered in issue #396, but was fixed between 0.0.21 and 0.0.23